torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -132,7 +132,90 @@ FunctionParameter::FunctionParameter(const std::string& fmt, bool keyword_only)
   }
 }
 
-bool FunctionParameter::check(PyObject* obj) {
+static PyObject* get_tensor_torch_function(void)
+{
+  PyObject* method = PyObject_GetAttrString((PyObject*)THPVariableClass, "__torch_function__");
+  assert(method != nullptr);
```

Comment:
The prevailing convention in the codebase is to not use assert, and to use `TORCH_INTERNAL_ASSERT` instead. (To verify: the `assert` here really means that if it's wrong, we've fucked up, is that right? I can't easily tell if this function is only called in cases where we have already checked that `__torch_function__` exists)

Reply:
This function is called only if `__torch|_function__` exists. 

I will switch to `TORCH_INTERNAL_ASSERT`.